lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Icarus.hs

```diff
@@ -151,6 +172,84 @@ generateKeyFromSeed
     -> IcarusKey 'RootK XPrv
 generateKeyFromSeed = unsafeGenerateKeyFromSeed
 
+-- | Hardware Ledger devices generates keys from mnemonic using a different
+-- approach (different from the rest of Cardano).
```

Comment:
The approach for generating the master key and chain code is called [SLIP-0010](https://github.com/satoshilabs/slips/blob/master/slip-0010.md), right?

Reply:
Damn. I've been my head through these documents for days, going from one to another and trying to figure out which one applied to the Hardware Ledger that I lost track of them at some point but, I think you're right!